src/App.tsx

```diff
@@ -1,61 +1,120 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(todoUserId: number) {
+  return usersFromServer.find(userId => userId.id === todoUserId);
+}
+
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosWithUsers);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError('Please enter title');
+
+      return;
```

Comment:
Because of this return, we can only see one error message at a time. We can try to combine this checks or create boolean flag to allow or disallow submit